airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/output/OutputConsumer.kt

```diff
@@ -90,38 +101,79 @@ interface OutputConsumer : Consumer<AirbyteMessage>, AutoCloseable {
     }
 }
 
+/** Configuration properties prefix for [StdoutOutputConsumer]. */
+const val CONNECTOR_OUTPUT_PREFIX = "airbyte.connector.output"
+
 // Used for integration tests.
-const val CONNECTOR_OUTPUT_FILE = "airbyte.connector.output.file"
+const val CONNECTOR_OUTPUT_FILE = "$CONNECTOR_OUTPUT_PREFIX.file"
 
 /** Default implementation of [OutputConsumer]. */
 @Singleton
 @Secondary
-private class StdoutOutputConsumer : OutputConsumer {
-    override val emittedAt: Instant = Instant.now()
+private class StdoutOutputConsumer(
+    val stdout: PrintStream,
```

Comment:
I'm not clear how this is being wired in. Does micronaut default PrintStream to ConsolePrintStream? Couldn't find any doc around that behavior. Not sure how this is related to record serialization opt

Reply:
Micronaut doesn't make those kinds of assumptions. To inject a non-bean instance it will look for a `@Factory`, in this case `PrintStreamFactory` defined further below.

Reply:
yeah my bad. Not sure how I missed it...